amaabca/ama_logger
# ADO 96835 Add execution_time to JSON formatter

[ADO 96835](https://dev.azure.com/AMA-Ent/AMA-Ent/_workitems/edit/96835)

### Changes
- Fetch new field `execution_time` of `float` type and map it to `executionTime` in the JSON formatter.
  - We will be able to use and filter `executionTime` in Loggly like [this query](https://ama.loggly.com/search#terms=tag:gatekeeper%20and%20json.eventName:%22login:benchmark%22&from=2024-02-27T18:40:00.515Z&until=2024-02-27T18:55:00.515Z&source_group=):
  ![image](https://github.com/amaabca/ama_logger/assets/7877/15b75e4d-4c31-4702-a8af-649fd4694418)
- Add a unit test.
- Bump the minor version.
- Git ignore the `.idea` folder created by Rubymine.

### Steps to test
- Check the Gatekeeper branch `96835-benchmark` is deployed to Gatekeeper staging in the #team-red-deployments Slack channel.
- Go to https://dev-my.ama.ab.ca/ and log in using an existing test account that you have. Or search for a random member in [iMIS UAT](https://crp-imisap-vq03.corp.ads/IMISUAT) and create a new account.
- Run [this Loggly query](https://ama.loggly.com/search#terms=tag:gatekeeper%20and%20json.eventName:%22login:benchmark%22&from=-10m&until=now&source_group=) and see the `login:benchmark` events generated from your logins.
- See the min, max, mean statistics in the left panel and do some filtering to see logins that took more than a certain amount of time.
